src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -85,8 +83,12 @@ public static function getSubscribedEvents(): array
         return [KernelEvents::CONTROLLER_ARGUMENTS => ['onKernelControllerArguments', 20]];
     }
 
-    private function getIsGrantedSubject(string|Expression $subjectRef, Request $request, array $arguments): mixed
+    private function getIsGrantedSubject(string|Expression|\Closure $subjectRef, Request $request, array $arguments): mixed
     {
+        if ($subjectRef instanceof \Closure) {
+            return $subjectRef($arguments, $request);
```

Comment:
This change means we now support closures nested inside an array as well. Is this expected ?

Reply:
I think that this was a flaw in the previous implementation, so yes, it's expected to me.